utils/nctl/sh/scenarios/common/itst.sh

```diff
@@ -125,6 +125,72 @@ function do_await_n_blocks() {
     nctl-await-n-blocks offset="$BLOCK_COUNT"
 }
 
+function _verify_deploy_transfer() {
+    local NODE_ID=${1}
+    local TARGET_HASH=${2}
+
+    # What kind of hash we are looking for, can be "transfer" or "deploy".
+    local TYPE=${3}
+
+    # Number of blocks to walkback before erroring out.
+    local WALKBACK=${4}
+
+    # Starting block hash to begin walking back from.
+    local BLOCK_HASH=${5}
+
+    local JSON_OUT
+    local PARENT
+    local HASHES
+    local BLOCK_HEADER
+
+    if [ -z "$BLOCK_HASH" ]; then
+        JSON_OUT=$($(get_path_to_client) get-block --node-address $(get_node_address_rpc "$NODE_ID"))
+        BLOCK_HASH=$(echo "$JSON_OUT" | jq -r '.result.block.hash')
+    else
+        JSON_OUT=$($(get_path_to_client) get-block --node-address $(get_node_address_rpc "$NODE_ID") -b "$BLOCK_HASH")
+    fi
+
+    BLOCK_HEADER=$(echo "$JSON_OUT" | jq '.result.block.header')
+
+    if [ "$WALKBACK" -gt 0 ]; then
+        HASHES=$(echo "$JSON_OUT" | jq ".result.block.body.${TYPE}_hashes")
+
+        if [ "$(echo "$HASHES" | jq "index( \"$TARGET_HASH\" )")" = "null" ]; then
+            PARENT=$(echo "$BLOCK_HEADER" | jq -r '.parent_hash')
+            WALKBACK=$((WALKBACK - 1))
+            log "$WALKBACK: Walking back to block: $PARENT"
+            _verify_deploy_transfer "$NODE_ID" "$TARGET_HASH" "$TYPE" "$WALKBACK" "$PARENT"
+        else
+            log "$TYPE verified, found in block: $BLOCK_HASH"
+        fi
+    else
+        log "Error: Block with $TYPE hash not found within walkback range!"
+        exit 1
+    fi
+}
+
+# Walks the chain on a given node, looking for a specific transfer hash.
+# Prints out the block hash that contains the target hash, if found.
+function verify_transfer() {
+    local NODE_ID=${1}
+    local TRANSFER_HASH=${2}
+    local WALKBACK=${3}
+    local BLOCK_HASH=${4}
```

Comment:
The method says that it prints out the block hash that contains the transfer we're looking for, what's _this_ `BLOCK_HASH` argument then?

Second question, can we also return the hash of the block containing the transfer? It might be useful to have downstream.